loki/transformations/inline.py

```diff
@@ -618,13 +618,27 @@ def inline_marked_subroutines(routine, allowed_aliases=None, adjust_imports=True
 
                 # If we're importing the same module, check for missing symbols
                 if m := imported_module_map.get(impt.module):
-                    if not all(s in m.symbols for s in impt.symbols):
+                    _m = import_map.get(m, m)
+                    if not all(s in _m.symbols for s in impt.symbols):
                         new_symbols = tuple(s.rescope(routine) for s in impt.symbols)
-                        import_map[m] = m.clone(symbols=tuple(set(m.symbols + new_symbols)))
+                        import_map[m] = m.clone(symbols=tuple(set(_m.symbols + new_symbols)))
 
         # Finally, apply the import remapping
         routine.spec = Transformer(import_map).visit(routine.spec)
 
+        # Add missing explicit interfaces from inlined subroutines
+        new_intfs = []
+        intf_symbols = routine.interface_symbols
+        for callee in call_sets.keys():
+            for intf in callee.interfaces:
+                for b in intf.body:
+                    s = getattr(b, 'procedure_symbol', None)
```

Comment:
Can we not use `intf.symbols` here?

Reply:
`Interface.body` comprises of `Subroutine` objects and not just the symbols, so I did it like this to capture the whole `Subroutine` object. If using `intf.symbols`, I would have to add `symbol.type.dtype.procedure` to the body rather than the symbol. Either is fine by me, please let me know if you would like me to update to the latter.

Reply:
Yes, I'm aware - but you're only inspecting the procedure symbols of these subroutines not the objects themselves, which is what `intf.symbols` should give you. Have a look at the implementation of `Interface.symbols`, which should be essentially the same as what you do here, but also includes the case of named interfaces etc: https://github.com/ecmwf-ifs/loki/blob/77114a954c0d774831d5ab7d4138774adaa20aba/loki/ir/nodes.py#L700-L710